src/picongpu/include/plugins/adios/ADIOSWriter.hpp

```diff
@@ -768,21 +769,25 @@ class ADIOSWriter : public ILightweightPlugin
 
         __getTransactionEvent().waitForFinished();
 
-        mThreadParams.adiosFilename = filename;
-        if (isCheckpoint)
+        std::string bpFilename( filename );
+        std::string bpFiledir( outputDirectory );
+        if( isCheckpoint )
         {
-            /* if checkpointFilename is relative, prepend with checkpointDirectory */
-            if (!boost::filesystem::path(checkpointFilename).has_root_path())
-                mThreadParams.adiosFilename = checkpointDirectory + std::string("/") + checkpointFilename;
-            else
-                mThreadParams.adiosFilename = checkpointFilename;
+            bpFilename = checkpointFilename;
+            bpFiledir = checkpointDirectory;
+        }
+
+        /* if file name is relative, prepend with common directory */
+        if( boost::filesystem::path(filename).has_root_path() )
```

Comment:
> Why is there no check for `isCheckpoint` anymore?

because I decluttered the if-casade-madness.

>  Why do you use filename here if you could use bpFilename?

that's a bug! will update!